x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/EnterpriseSearch.java

```diff
@@ -131,8 +132,13 @@ public Collection<Object> createComponents(
             xContentRegistry
         );
         analyticsTemplateRegistry.initialize();
+        final AnalyticsCollectionService analyticsCollectionService = new AnalyticsCollectionService(
```

Comment:
ℹ️ Maybe the service do not need to be created here and can just be injected where needed. I will reevaluate this aspect before making the PR ready for reviews.

Reply:
It is required to init the `AnalyticsCollectionService` during the plugin start since it is a `ClusterStateListener`.